src/Common/src/System/Net/WebSockets/WebSocketValidate.cs

```diff
@@ -0,0 +1,111 @@
+                throw new ArgumentException(SR.net_WebSockets_InvalidEmptySubProtocol, "subProtocol");
+            }
+
+            string invalidChar = null;
+            int i = 0;
+            while (i < subProtocol.Length)
+            {
+                char ch = subProtocol[i];
+                if (ch < 0x21 || ch > 0x7e)
+                {
+                    invalidChar = string.Format(CultureInfo.InvariantCulture, "[{0}]", (int)ch);
+                    break;
+                }
+
+                if (!char.IsLetterOrDigit(ch) &&
+                    Separators.IndexOf(ch) >= 0)
+                {
+                    invalidChar = ch.ToString();
+                    break;
+                }
+
+                i++;
+            }
+
+            if (invalidChar != null)
+            {
+                throw new ArgumentException(SR.Format(SR.net_WebSockets_InvalidCharInProtocolString, subProtocol, invalidChar),
+                    "subProtocol");
+            }
+        }
+
+        internal static void ValidateCloseStatus(WebSocketCloseStatus closeStatus, string statusDescription)
+        {
+            if (closeStatus == WebSocketCloseStatus.Empty && !string.IsNullOrEmpty(statusDescription))
+            {
+                throw new ArgumentException(SR.Format(SR.net_WebSockets_ReasonNotNull,
+                    statusDescription,
+                    WebSocketCloseStatus.Empty),
+                    "statusDescription");
+            }
+
+            int closeStatusCode = (int)closeStatus;
+
+            if ((closeStatusCode >= InvalidCloseStatusCodesFrom &&
+                closeStatusCode <= InvalidCloseStatusCodesTo) ||
+                closeStatusCode == CloseStatusCodeAbort ||
+                closeStatusCode == CloseStatusCodeFailedTLSHandshake)
+            {
+                // CloseStatus 1006 means Aborted - this will never appear on the wire and is reflected by calling WebSocket.Abort
+                throw new ArgumentException(SR.Format(SR.net_WebSockets_InvalidCloseStatusCode,
+                    closeStatusCode),
+                    "closeStatus");
+            }
+
+            int length = 0;
+            if (!string.IsNullOrEmpty(statusDescription))
+            {
+                length = Encoding.UTF8.GetByteCount(statusDescription);
+            }
+
+            if (length > WebSocketValidate.MaxControlFramePayloadLength)
+            {
+                throw new ArgumentException(SR.Format(SR.net_WebSockets_InvalidCloseStatusDescription,
+                    statusDescription,
+                    WebSocketValidate.MaxControlFramePayloadLength),
+                    "statusDescription");
+            }
+        }
+
+        internal static void ThrowPlatformNotSupportedException()
+        {
+            throw new PlatformNotSupportedException(SR.net_WebSockets_UnsupportedPlatform);
+        }
+
+        internal static void ValidateArraySegment<T>(ArraySegment<T> arraySegment, string parameterName)
```

Comment:
I was curious about the ".Array" being added to the parameter name, and so searched for where this function is used... but couldn't find any call sites in this PR.  Is it used somewhere?